src/main/kotlin/exercise5/BoardImpl.kt

```diff
@@ -1,13 +1,93 @@
 package exercise5
 
-// TODO Instantiate SquareBoard
-fun createSquareBoard(width: Int): SquareBoard = TODO("SquareBoardImpl(width)")
+fun createSquareBoard(width: Int): SquareBoard = SquareBoardImpl(width)
 
-// TODO Instantiate GameBoard
-fun <T> createGameBoard(width: Int): GameBoard = TODO("GameBoardImpl(width)")
+fun <T> createGameBoard(width: Int): GameBoard = GameBoardImpl(width)
 
-// TODO Implement SquareBoard Interface
-class SquareBoardImpl(size: Int)
+open class SquareBoardImpl(size: Int) : SquareBoard {
+    protected val board = List(size) { x ->
+        List(size) { y ->
+            Cell(x + 1, y + 1)
+        }
+    }
 
-// TODO extend SquareBoardImpl and implement GameBoard interface
-class GameBoardImpl(size: Int)
\ No newline at end of file
+    override val width = size
+
+    override fun getCellOrNull(i: Int, j: Int): Cell? {
+        if(i < 1 || i > width || j < 1 || j > width) return null
+
+        return board[i - 1][j - 1]
+    }
+
+    override fun getCell(i: Int, j: Int): Cell {
+        val cell = getCellOrNull(i, j)
+        require(cell != null) {
+            throw IllegalArgumentException("Wrong parameter!")
+        }
+
+        return cell
+    }
+
+    override fun getAllCells(): Collection<Cell> {
+        return board.flatten()
+    }
+
+    override fun getRow(i: Int, jRange: IntProgression): List<Cell> {
+        return jRange.filter { it in 1..width }.
+                      map { getCell(i, it) }
+    }
+
+    override fun getColumn(iRange: IntProgression, j: Int): List<Cell> {
+        return iRange.filter { it in 1..width }.
+                      map { getCell(it, j) }
+    }
+
+    override fun Cell.getNeighbour(direction: Direction): Cell? {
+        return when(direction) {
+            Direction.UP -> getCellOrNull(i - 1, j)
+            Direction.DOWN -> getCellOrNull(i + 1, j)
+            Direction.RIGHT -> getCellOrNull(i, j + 1)
+            Direction.LEFT -> getCellOrNull(i, j - 1)
+        }
+    }
+
+
+}
+
+class GameBoardImpl(size: Int) : SquareBoardImpl(size), GameBoard {
+
+    private val map = mutableMapOf<Cell, String?>()
+
+    init {
+        board.flatten().forEach {
+            map[it] = null
+        }
+    }
```

Comment:
Also can be simplified as following :) 
```suggestion
    private val map = board.flatten().map { it to null }.toMutableMap()
```